sympy/functions/combinatorial/factorials.py

```diff
@@ -1100,3 +1110,107 @@ def _eval_is_nonnegative(self):
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
         from sympy.functions.special.gamma_functions import gamma
         return self.rewrite(gamma)._eval_as_leading_term(x, logx=logx, cdir=cdir)
+
+
+def _binomial_modulo_p_pow(n, m, p, q):
+    """Compute binomial(n, m) % p^q.
+
+    Parameters
+    ==========
+    n : positive integer
+    m : a nonnegative integer
+    p : a prime
+    q : a positive integer (the prime exponent)
+
+    References
+    ==========
+
+    .. [1] Binomial coefficients modulo prime powers, [online],
```

Comment:
Can you format this according to the guide at https://docs.sympy.org/latest/contributing/docstring.html#references-section (at the very least include the author name here)

Reply:
Apologies, I've included the author's name in the reference.  I'm not sure reference citation/journal/etc apply here since I'm not sure if/where this paper has been published - from what I can see on google scholar.